File: 151003/Haritonenko/task345/publisher/src/main/java/by/bsuir/publisher/dto/requests/converters/MarkerRequestConverter.java
package by.bsuir.publisher.dto.requests.converters;

import by.bsuir.publisher.domain.Marker;
import by.bsuir.publisher.dto.requests.MarkerRequestDto;
import org.mapstruct.Mapper;

@Mapper(componentModel = "spring")
public interface MarkerRequestConverter {
    Marker fromDto(MarkerRequestDto marker);
}

File: 151003/Yagnnish/Lab1/Dtos/RequestDtos/RequestConverters/MarkerRequestConverter.cs
using lab_1.Domain;

namespace lab_1.Dtos.RequestDtos.RequestConverters
{
    public class MarkerRequestConverter : BaseRequest<Marker, MarkerRequestDto>
    {
        public Marker FromDto(MarkerRequestDto dto, long? id)
        {
            return new Marker(id, dto.name);
        }
    }
}
